todo_app/app.py

```diff
@@ -8,4 +8,77 @@
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    if "todoes" in session:
+        if "removed" in session:
+            return render_template("index.html",todos=session["todoes"],added=session["added"],complete=session["complete"],removed=session["removed"])
+        else:
+            return render_template("index.html",todos=session["todoes"],added=session["added"])
+    else:
+        return render_template("index.html")
```

Comment:
This code is getting pretty complex to read already! You seem to handle the case that `session["complete"]` and `session["removed"]` are missing without issues, can we do the same for the others? One way would be to take advantage of the `.get(key, default)` syntax available, to get an empty array if any value is missing:

```suggestion
    return render_template(
        "index.html",
        todos=session.get("todoes", []),
        added=session.get("added", []),
        complete=session.get("complete", []),
        removed=session.get("removed", [])
    )
```